### PR TITLE
Update token exchange environment variable name

### PR DESCRIPTION
**Pull request by chlowell**
TOKEN_FILE_PATH -> AZURE_FEDERATED_TOKEN_FILE

FYI @christothes @g2vinay @sadasant

### CODE REVIEW

File: sdk/identity/azure-identity/azure/identity/_constants.py [hunk 1]
```diff
@@ -47,5 +47,5 @@ class EnvironmentVariables:
     AZURE_IDENTITY_ENABLE_LEGACY_TENANT_SELECTION = "AZURE_IDENTITY_ENABLE_LEGACY_TENANT_SELECTION"
     AZURE_REGIONAL_AUTHORITY_NAME = "AZURE_REGIONAL_AUTHORITY_NAME"
 
-    TOKEN_FILE_PATH = "TOKEN_FILE_PATH"
-    TOKEN_EXCHANGE_VARS = (AZURE_CLIENT_ID, AZURE_TENANT_ID, TOKEN_FILE_PATH)
+    AZURE_FEDERATED_TOKEN_FILE = "AZURE_FEDERATED_TOKEN_FILE"
```
[the remark this answers is not in the document]
Comment (chlowell) [reply sdk/identity/azure-identity/azure/identity/_constants.py:50]: By "be less breaking" do you mean "replacing fewer references" (there's no API to break, this is an internal constant)? Having the constant match its value seems more important to me.

[the remark this answers is not in the document]
Comment (xiangyan99) [reply sdk/identity/azure-identity/azure/identity/_constants.py:50]: My understanding is if we keep TOKEN_FILE_PATH , we don't need to update those in tests?

[the remark this answers is not in the document]
Comment (chlowell) [reply sdk/identity/azure-identity/azure/identity/_constants.py:50]: That makes the diff of this PR smaller, but then the constant doesn't match its value. Given its meaning and how it's used, that seems more important to me.

[the remark this answers is not in the document]
Comment (xiangyan99) [reply sdk/identity/azure-identity/azure/identity/_constants.py:50]: Sounds fair.

**Review by xiangyan99 — APPROVED** [review]

**Review by mccoyp — APPROVED** [review]

**Pull request merged**